components/match2/wikis/overwatch/match_group_input_custom.lua

```diff
@@ -0,0 +1,542 @@
+--
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+
+local _ALLOWED_STATUSES = { 'W', 'FF', 'DQ', 'L', 'D' }
+local _MAX_NUM_OPPONENTS = 8
+local _MAX_NUM_PLAYERS = 10
+local _MAX_NUM_VODGAMES = 9
+local _MAX_NUM_MAPS = 9
+local _DEFAULT_BESTOF = 3
+local _NO_SCORE = -99
+
+local _EPOCH_TIME = '1970-01-01 00:00:00'
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+local opponentFunctions = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(_, match)
+	-- Count number of maps, and automatically count score
+	match = matchFunctions.getBestOf(match)
+	match = matchFunctions.getScoreFromMapWinners(match)
+
+	-- process match
+	Table.mergeInto(
+		match,
+		matchFunctions.readDate(match)
+	)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+	match = matchFunctions.getExtraData(match)
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processMap(_, map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+	map = mapFunctions.getTournamentVars(map)
+
+	return map
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processOpponent(_, opponent)
+	-- check for empty opponent and convert to literal
+	if type(opponent) == 'table' and opponent.type == 'team' and Logic.isEmpty(opponent.template) then
+		opponent.name = ''
+		opponent.type = 'literal'
+	end
+
+	-- check for lazy bye's and convert them to literals
+	if type(opponent) == 'table' and string.lower(opponent.template or '') == 'bye' then
+			opponent.name = 'BYE'
+			opponent.type = 'literal'
+	end
+
+	return opponent
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processPlayer(_, player)
+	return player
+end
+
+--
+--
+-- function to check for draws
+function CustomMatchGroupInput.placementCheckDraw(table)
+	local last
+	for _, scoreInfo in pairs(table) do
+		if scoreInfo.status ~= 'S' and scoreInfo.status ~= 'D' then
+			return false
+		end
+		if last and last ~= scoreInfo.score then
+			return false
+		else
+			last = scoreInfo.score
+		end
+	end
+
+	return true
+end
+
+function CustomMatchGroupInput.getResultTypeAndWinner(data, indexedScores)
+	-- Map or Match wasn't played, set not played
+	if
+		data.finished == 'skip' or
+		data.finished == 'np' or
+		data.finished == 'cancelled' or
+		data.finished == 'canceled' or
+		data.winner == 'skip' or
+		data.winner == 'np' or
+		data.winner == 'cancelled' or
+		data.winner == 'canceled'
+	then
```

Comment:
`Table.includes`

Reply:
'Table.includes' where? From what I can tell that isn't used in this function on most of the other Match2 MatchGroup/Input modules

Reply:
set at the top something like
```lua
local _FINISHED_INDICATORS = {
	'skip',
	'np',
	'cancelled',
	'canceled',
}
```
then at when checking it:
```suggestion
	if Table.includes(_FINISHED_INDICATORS, data.finished) or Table.includes(_FINISHED_INDICATORS, data.winner) then
```

Reply:
Keep the variable same name as on Mobile Legends though